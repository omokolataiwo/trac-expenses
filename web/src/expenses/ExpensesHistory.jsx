import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledExpenses = styled.div`
  display: flex;
`;

const ExpensesHistory = ({ expenses }) => (
  <div>
    <h2>Expenses History</h2>
    {expenses.map(expense => (
      <StyledExpenses>
        <span>{expense.date.toString()}</span>
        <span>{expense.description}</span>
        <span>{expense.amount}</span>
      </StyledExpenses>
    ))}
  </div>
);

ExpensesHistory.propTypes = {
  expenses: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      amount: PropTypes.string
    })
  ).isRequired
};
export default ExpensesHistory;
