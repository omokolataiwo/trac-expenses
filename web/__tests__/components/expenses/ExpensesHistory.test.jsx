import React from 'react';
import { shallow } from 'enzyme';
import ExpensesHistory from '../../../src/expenses/ExpensesHistory';

const props = {
  expenses: [{ description: 'Tranportation to office', amount: '1000', date: 'Tue. August 14 2019' }]
};

const wrapper = shallow(<ExpensesHistory {...props} />);

describe('ExpensesHistory Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
