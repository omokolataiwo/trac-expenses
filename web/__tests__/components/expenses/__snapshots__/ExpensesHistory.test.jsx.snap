// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesHistory Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesHistory
    expenses={
      Array [
        Object {
          "amount": "1000",
          "date": "Tue. August 14 2019",
          "description": "Tranportation to office",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Expenses History
        </h2>,
        Array [
          <ForwardRef(styled.div)>
            <span>
              Tue. August 14 2019
            </span>
            <span>
              Tranportation to office
            </span>
            <span>
              1000
            </span>
          </ForwardRef(styled.div)>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expenses History",
        },
        "ref": null,
        "rendered": "Expenses History",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span>
              Tue. August 14 2019
            </span>,
            <span>
              Tranportation to office
            </span>,
            <span>
              1000
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tue. August 14 2019",
            },
            "ref": null,
            "rendered": "Tue. August 14 2019",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tranportation to office",
            },
            "ref": null,
            "rendered": "Tranportation to office",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1000",
            },
            "ref": null,
            "rendered": "1000",
            "type": "span",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Expenses History
          </h2>,
          Array [
            <ForwardRef(styled.div)>
              <span>
                Tue. August 14 2019
              </span>
              <span>
                Tranportation to office
              </span>
              <span>
                1000
              </span>
            </ForwardRef(styled.div)>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expenses History",
          },
          "ref": null,
          "rendered": "Expenses History",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span>
                Tue. August 14 2019
              </span>,
              <span>
                Tranportation to office
              </span>,
              <span>
                1000
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tue. August 14 2019",
              },
              "ref": null,
              "rendered": "Tue. August 14 2019",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tranportation to office",
              },
              "ref": null,
              "rendered": "Tranportation to office",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1000",
              },
              "ref": null,
              "rendered": "1000",
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
